src/lib/action.rs

```diff
@@ -35,29 +35,37 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-// #![warn(missing_docs)]
+#![warn(missing_docs)]
 
+use std::boxed::Box;
 use std::clone::Clone;
 use std::fmt;
 
 use ast::{Arena, ArenaError, ArenaResult, NodeId};
 
 /// Apply an action to an AST node.
-pub trait ApplyAction<T: Clone> {
+pub trait ApplyAction<T: Clone + fmt::Display>: fmt::Display {
+    /// Apply an action to an AST.
     fn apply(&mut self, arena: &mut Arena<T>) -> ArenaResult;
 }
 
 /// A list of actions to be applied.
 ///
 /// This type will usually be used by iterating over the list and calling
 /// `apply()` on each element.
-pub type ActionList<T> = Vec<Box<ApplyAction<T>>>;
+pub struct EditScript<T> {
+    actions: Vec<Box<ApplyAction<T>>>,
+}
 
+#[derive(Copy, Clone, Eq, PartialEq)]
 /// Delete a node from a given AST.
+///
+/// It is only valid to delete leaf nodes.
```

Comment:
So that it can implement the `ApplyAction` trait. This is from a previous PR, I think.